### ISSUE TITLE
Terraform destroy doesn't delete PTR record

### ISSUE THREAD

**Issue by andy812** [pan-net/terraform-provider-powerdns#49]
### Terraform Version
Terraform v0.12.17

### Affected Resource(s)
powerdns_record

### Terraform Configuration Files
```hcl
resource "powerdns_record" "dev-minio" {
  zone    = "example.com."
  name    = "dev-minio.example.com."
  type    = "A"
  ttl     = 360
  set_ptr = true
  records = ["172.20.30.14"]
}
```

### Expected Behavior
`terraform destroy` should delete A and PTR records

### Actual Behavior
`terraform destroy` deletes only A record

### Steps to Reproduce
1. `terraform apply` configuration above
Terraform creates A record and PTR record (because `set_ptr = true`)
2. `terraform destroy`
Terraform deletes only A record.

**Comment by splashx** [comment]
@andy812 what happens if you delete the record via API? Does PowerDNS remove the record?
I ask because the `set_ptr: true` option is a powerdns feature (not a provider per-se feature) - the automatic creation of the PTR is done automagically by PowerDNS (so one should expect that the opposite is also true: when deleting, PowerDNS should automagically delete the PTR).

**Comment by andy812** [comment]
> @andy812 what happens if you delete the record via API directly? Does PowerDNS remove the PTR record?
No, i shoud make 2 API calls: delete A record, then delete PTR record.
I expect what provider will make 2 API calls too when i run `terraform destroy` and `set_ptr = true` is in tfstate.
The main problem is there is no information about PTR record in tfstate.

**Comment by mbag** [comment]
As @splashx  mentions the `set_ptr` logic is implemented on the side of the PowerDNS API server. There was a proposal in PowerDNS repo to create `delete-ptr` option, so that PTR record would get deleted automatically, here is the feature request: https://github.com/PowerDNS/pdns/issues/4656
When you read whole thread, the conclusion was that not only will `delete-ptr` not be implemented, but they are deprecating `set-ptr` in v4.3.0 of the PowerDNS. This means we will be deprecating it as well.

I have to check if PowerDNS API returns information about PTR record it created when set-ptr is present in request. Unless API returns this info, I'm afraid there is no way of storing that information in tfstate. Terraform providers should be viewed as thin clients for an underlaying API, not as something that should implemet missing features of the API.

**Comment by splashx** [comment]
@mbag first course of action IMHO is to update the documentation to explicitly warn users about what terraform does or does not when `set_ptr = true` (so, at this moment, write that if this is set to `true`, terraform *will not delete* the PTR record). Users should at least know that - otherwise expectations differ.

**Comment by splashx** [comment]
I would also add in the documentation that users are advised to manage their PTR records explicitly - zone creation and record type `PTR` are supported by the provider. Although not ideal, this is anyway what is going to happen from 4.3, so we better get ready for that..

**Comment by anuriq** [comment]
It would be great if folks from PowerDNS contributed their 2 cents to this provider. It seems a lot of people use this to manage their PowerDNS servers.

**Comment by mbag** [comment]
I have created PR making changes to documentation: https://github.com/terraform-providers/terraform-provider-powerdns/pull/50 
I appreciate comments and notes for improvement. Once merged, the changes to documentation will propagate fairly quickly, since Hashicorp builds terraform website docs fairly often. 

I will now take a look at whether I can store PTR in tfstate, and also how to refuse this attribute if provider detects it's talking to PowerDNS API v4.3.0

**Comment by mbag** [comment]
I tried creating A record with `set-ptr: true` against PowerDNS test server created using docker-compose in this repo (started locally using `docker-compose run --rm setup`). API server is initialized with reverse zone `in-addr.arpa.`

I used following `curl` command, but server returned no information about PTR record, that could be stored in tfstate, to track what PTR record should be deleted.
```bash
curl -X PATCH --data '{"rrsets": [ {"name": "test.sysa.xyz.", "type": "A", "ttl": 400, "changetype": "REPLACE", "records": [ {"content": "1.1.1.1", "disabled": false, "set-ptr": true } ] } ] }' -H 'X-API-Key: secret' http://localhost:8081/api/v1/servers/localhost/zones/sysa.xyz.
```

I was curios so I did another test. I created another reverse zone `1.1.in-addr.arpa`, that was more specific for A records with IP addresses starting with `1.1.X.X` and ran following `curl` command:
```bash
curl -X PATCH --data '{"rrsets": [ {"name": "test2.sysa.xyz.", "type": "A", "ttl": 400, "changetype": "REPLACE", "records": [ {"content": "1.1.1.2", "disabled": false, "set-ptr": true } ] } ] }' -H 'X-API-Key: secret' http://localhost:8081/api/v1/servers/localhost/zones/sysa.xyz.
```
Again nothing about PTR record was returned from server and PTR record ended up created in `1.1.in-addr.arpa` zone, as it was more specific than the `in-addr.arpa` for given IP address.

So to implement PTR deletion which was created automatically via `set-ptr` option, we would need to find a zone in which PTR record is located by checking `in-addr.arpa.`, and other subzones based on IP address. I'll look into code if it would be possible to implement this logic without breaking logic of the provider.
